contracts/MarketCollateralised.sol

```diff
@@ -68,44 +68,60 @@ abstract contract MarketCollateralised is Market {
+		uint256 existingMarketCollateral = _marketCollateral[marketId] + _marketTotal[marketId];
+
+		// If we have enough collateral in this market already, we're done	
+		if (_potentialPayout[propositionId] <= existingMarketCollateral) {
+			return 0;
+		}
 		
-		if (_potentialPayout[propositionId] > existingCollateral) {
-			// Get any additional collateral we need for this market
-			_mostExpensivePropositionId[marketId] = propositionId;
-			result = _potentialPayout[propositionId] - existingCollateral;
-			if (_totalCollateral > _totalExposure) {
-				// We have some collateral available to use (from previous bets
-				uint256 internallyAvailableCollateral = _totalCollateral - _totalExposure;
-				internalCollateralToUse = Math.min(result, internallyAvailableCollateral);
-				_totalExposure += internalCollateralToUse;
-			}
-			if (internalCollateralToUse < result) {
-				// We need to get more collateral from the Vault
-				result = result - internalCollateralToUse;
-				IERC20(_vault.asset()).transferFrom(
-					address(_vault),
-					_self,
-					result
-				);
-				
-				emit Borrowed(index, result);
-
-				_totalCollateral += result;
-			}
-			_marketCollateral[marketId] += result;
+		// Otherwise, this must be the new "most expensive" proposition and we need to obtain more collateral to cover the gap.
+		_mostExpensivePropositionId[marketId] = propositionId;
+		
+		// The amount needed is the difference between the potential payout and the existing collateral on this race
+		uint256 amountToObtain = _potentialPayout[propositionId] - existingMarketCollateral;
+		
+		// If the total amount of collateral in the contracts is greater than the amount backing bets, we can use some
+		uint256 internalCollateralToUse;
+		if (_totalCollateral > _totalExposure) {
+			// We have some collateral available to use (from previous bets)
+			// This is the total collateral not already covering bets
+			uint256 internallyAvailableCollateral = _totalCollateral - _totalExposure;
+
+			// We can only use the amount of collateral that is available
+			internalCollateralToUse = Math.min(amountToObtain, internallyAvailableCollateral);
 		}
 
-		_betExposure[_bets.length] = result;
-		return result;
+		// If the amount we need to borrow is greater than the amount of collateral we can use, we need to get more collateral from the Vault
+		if (internalCollateralToUse < amountToObtain) {
+			// We need to get more collateral from the Vault
+			uint256 amountToBorrow = amountToObtain - internalCollateralToUse;
+			IERC20(_vault.asset()).transferFrom(
+				address(_vault),
+				_self,
+				amountToBorrow
+			);
+
+			// Add the amount we borrowed to the total collateral				
+			emit Borrowed(index, amountToBorrow);
+			_totalCollateral += amountToBorrow;
+		}
+
+		// Store the amount of exposure that this bet added. This will be deducted from the total exposure when the bet is paid out
+		_betExposure[_bets.length] = amountToObtain;
```

Comment:
quick q on why we map the `_betExposure` by the total bets and not the bet `index` ?

Reply:
Because the bet hasn't been added the the array yet

Reply:
I see, we actually do have the `index` there! I'll change it.